container_pipeline/workers/base.py

```diff
@@ -27,6 +32,49 @@ def handle_job(self, job):
         """
         raise NotImplementedError
 
+    def setup_data(self):
+        self.build = Build.objects.get(uuid=self.job['uuid'])
+        self.build_phase = BuildPhase.objects.get(
+            build=self.build, phase=self.build_phase_name)
+
+    def set_buildphase_data(self, build_phase_status=None,
+                            build_phase_start_time=None,
+                            build_phase_end_time=None,
+                            build_phase_log_file=None):
+
+        if build_phase_status:
+            self.build_phase.status = build_phase_status
+        if build_phase_start_time:
+            self.build_phase.start_time = build_phase_start_time
+        if build_phase_end_time:
+            self.build_phase.end_time = build_phase_end_time
+        if build_phase_log_file:
+            self.build_phase.log_file_path = build_phase_log_file
+        self.build_phase.save()
+
+    def init_next_phase_data(self, next_phase_name):
+        next_phase, created = BuildPhase.objects.get_or_create(
+            build=self.build, phase=next_phase_name)
+        next_phase.status = 'queued'
+        next_phase.save()
+
+    def set_build_data(self, build_status=None,
+                       build_end_time=None, build_trigger=None):
+
+        if build_status:
+            self.build.status = build_status
```

Comment:
This is not the same as `build_status` we have in the `job`, right? The one in `job` dictionary is a bool.

Reply:
correct.